### PR TITLE
Support allowCachedMaxAge in js-sdk

### PR DESCRIPTION
**Pull request by yolken-airplane**
## Description
This change updates the task config schema in the js-sdk to support setting `allowMaxCachedAge`. It also adds a `allowCachedMaxAge ` client option so that this can be customized on a per-task-execution basis. 

See https://www.notion.so/Server-side-task-caching-a1e78a9b426e4456929827a05437a440 for more details.

## Testing
Testing completed successfully via deploying locally.

### CODE REVIEW

**Review by ameyakhare — APPROVED** [review]

**Comment by yolken-airplane** [comment]
@ameyakhare updated this change to also allow setting `allowCachedMaxAge` in execute calls. Can you take another look? Thanks!

File: packages/airplane/internal/api/client.ts [hunk 1]
```diff
@@ -142,6 +146,9 @@ export class Client {
             slug: slug,
             paramValues: paramValues ?? {},
             resources: resources ?? {},
+            ...(typeof this.allowCachedMaxAge !== "undefined" && {
```
Comment (yolken-airplane) [inline packages/airplane/internal/api/client.ts:149]: Not using `this.allowCachedMaxAge && ...` since we want to set the value if it's `0`.

Comment (ameyakhare) [reply packages/airplane/internal/api/client.ts:149]: why do you have to do typeof? can't you just check to see if the value is undefined or null?

**Review by ameyakhare — COMMENTED** [review]
1 comment

File: packages/airplane/internal/api/client.ts (hunk 1 above)
Comment (yolken-airplane) [reply packages/airplane/internal/api/client.ts:149]: That's what stackoverflow said :-). In any case, I can just set it directly.

Comment (yolken-airplane) [reply packages/airplane/internal/api/client.ts]: Done.

**Pull request merged**